source/extensions/common/wasm/wasm.cc

```diff
@@ -2110,7 +2110,7 @@ void Wasm::establishEnvironment() {
 
     global_table_base_ = wasm_vm_->makeGlobal("env", "__table_base", Word(0));
     global_dynamictop_ =
-        wasm_vm_->makeGlobal("env", "DYNAMICTOP_PTR", Word(emscripten_dynamictop_ptr_));
+        wasm_vm_->makeGlobal("env", "DYNAMICTOP_PTR", Word(emscripten_dynamic_base_));
```

Comment:
I don't think this is correct, [`DYNAMICTOP_PTR`'s address is encoded in `emscripten_dynamictop_ptr_`](https://github.com/emscripten-core/emscripten/blob/f715b7de7426bb27982c4fdeb94b5a32966781e1/tools/shared.py#L3076), and not in `emscripten_dynamic_base_`, which represents `DYNAMIC_BASE`.

Is this still necessary after your other changes? Or does this simply align with what WAVM is doing internally (which may or may not be correct, since that code predates Emscripten metadata, AFAIK).

Reply:
Done.